packages/utils/src/array/mapRight/index.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * @description 주어진 배열의 각 요소에 대해 `오른쪽에서 왼쪽으로` 순회하며 제공된 `콜백 함수`를 호출하고, 결과를 새로운 배열로 반환하는 함수입니다.
+ *
+ * @template T - 배열 요소의 유형입니다.
+ * @template U - 반환 배열 요소의 유형입니다.
+ * @param {T[] | readonly T[]} array - 순회할 배열입니다. 변경 가능한 배열 또는 읽기 전용 배열일 수 있습니다.
+ * @param {(currentValue: T, index: number, array: T[] | readonly T[]) => U} callback - 배열의 각 요소에 대해 호출할 함수입니다.
+ * - `currentValue` - 현재 처리 중인 배열 요소입니다.
+ * - `index` - 현재 요소의 인덱스입니다.
+ * - `array` - 순회 중인 배열 자체입니다.
+ * @param {any} [thisArg] - `callback`을 호출할 때 `this`로 사용할 값입니다.
+ * @returns {U[]} 새로운 배열을 반환합니다. 각 요소는 `callback`의 반환 값입니다.
+ */
+export const mapRight = <T, U>(
+  array: T[] | readonly T[],
+  callback: (currentValue: T, index: number, array: T[] | readonly T[]) => U,
+  thisArg?: any
+): U[] => {
+  const result: U[] = [];
+  for (let i = array.length - 1; i >= 0; i--) {
+    result.push(callback.call(thisArg, array[i], i, array));
+  }
+  return result;
+};
```

Comment:
```suggestion
export function mapRight <T, U>(
  array: T[] | readonly T[],
  callback: (currentValue: T, index: number, array: T[] | readonly T[]) => U,
): U[] {
  const result: U[] = [];
  for (let i = array.length - 1; i >= 0; i--) {
    result.push(callback(array[i], i, array));
  }
  return result;
};
```
기존 js map 인터페이스에 맞게 `thisArg`를 추가해주신 걸로 이해했는데, 해당 프로퍼티는 일반적인 상황에서 많이 사용하지 않고, 오히려 제공하고자 하는 mapRight의 인터페이스를 복잡하게 만들기 때문에 thisArg는 제거하면 어떨까 싶습니다 

추가적으로 함수 선언문으로 변경해주시면 감사드립낟

Reply:
좋습니다! 감사합니다.